boards/posix/common/sdl/CMakeLists.txt

```
@@ -7,6 +7,10 @@ zephyr_library_compile_definitions(NO_POSIX_CHEATS)
 find_package(PkgConfig REQUIRED)
 pkg_search_module(SDL2 REQUIRED sdl2)
 zephyr_include_directories(${SDL2_INCLUDE_DIRS})
-zephyr_link_libraries(${SDL2_LIBRARIES})
+if (CONFIG_NATIVE_APPLICATION)
+	zephyr_link_libraries(${SDL2_LIBRARIES})
+else()
+	target_link_options(native_simulator INTERFACE "-l${SDL2_LIBRARIES}")
```

minor nit, why using `target_link_options` when linking to a library ?

There is a `target_link_libraries(...)` available for that.
```suggestion
	target_link_libraries(native_simulator INTERFACE ${SDL2_LIBRARIES})
```

---

Discussed offline with @tejlmand :
`target_link_libraries()` does not seem directly/easily usable.
I will add a small comment in this cmake file in a follow up PR.
The reason for using target_link_options instead of target_link_libraries is that
target_link_libraries() updates INTERFACE_LINK_LIBRARIES but what ends up in it is something like `::@(0x56457c309b90) SDL2::@` (see https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_LINK_LIBRARIES.html#prop_tgt:INTERFACE_LINK_LIBRARIES ) which cannot be passed directly to the linker.
Right now we use target_link_options (and INTERFACE_LINK_OPTIONS) to pass and extract the links options for the runner final link.